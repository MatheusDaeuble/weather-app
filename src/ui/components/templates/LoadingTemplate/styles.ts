import styled from 'styled-components/native';
import { Spacing } from 'ui/theme';
import { colors } from 'ui/theme/colors';

export const Container = styled.View`
  flex: 1;
  width: 100%;
  height: 100%;
  justify-content: center;
  align-items: center;
  background: ${colors.primary};
  padding: ${Spacing.SCALE_28}px;
`;
