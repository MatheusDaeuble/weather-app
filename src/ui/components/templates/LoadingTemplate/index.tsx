import React from 'react';
import Loading from 'ui/components/atoms/Loading';
import SafeArea from 'ui/containers/SafeArea';
import { Container } from './styles';

const LoadingTemplate = () => {
  return (
    <SafeArea color='primary'>
      <Container>
        <Loading color='white' />
      </Container>
    </SafeArea>
  );
};

export default LoadingTemplate;
