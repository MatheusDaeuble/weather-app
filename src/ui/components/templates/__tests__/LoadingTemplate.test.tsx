import React from 'react';
import renderer from 'react-test-renderer';
import LoadingTemplate from '../LoadingTemplate';

describe('LoadingTemplate', () => {
  it('Default', () => {
    const component = renderer.create(<LoadingTemplate />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
