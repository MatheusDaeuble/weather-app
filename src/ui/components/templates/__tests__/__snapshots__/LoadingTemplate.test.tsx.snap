// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingTemplate Default 1`] = `
<RCTSafeAreaView
  color="primary"
  emulateUnlessSupported={true}
  isFlex={true}
  style={
    Array [
      Object {
        "backgroundColor": "#2E89DF",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#2E89DF",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "height": "100%",
          "justifyContent": "center",
          "paddingBottom": 56,
          "paddingLeft": 56,
          "paddingRight": 56,
          "paddingTop": 56,
          "width": "100%",
        },
      ]
    }
  >
    <View
      flex1={true}
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "justifyContent": "center",
          },
        ]
      }
    >
      <ActivityIndicator
        color="#FFFFFF"
        size="large"
      />
    </View>
  </View>
</RCTSafeAreaView>
`;
